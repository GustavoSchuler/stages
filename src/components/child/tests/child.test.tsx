import React from "react"
import { render, waitFor, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';

import Child from '../child.component';

describe('[Component] Child', () => {
  let wrapper: any;

  const mockedProps = {
    color: 'blue',
    duration: 1000,
    onLoaded: jest.fn(),
  }

  beforeEach(() => {
    jest.useFakeTimers();
    const { color, duration, onLoaded } = mockedProps;

    wrapper = render(
      <Child color={color} duration={duration} onLoaded={() => onLoaded} />
    );
  });

  afterEach(cleanup);

  it('renders correctly', () => {
    const element = wrapper.getByTestId('child__box');

    expect(element).toMatchSnapshot();
  });

  it('should render loader with correct color', () => {
    const element = wrapper.getByTestId('child__loader');

    expect(element).toHaveAttribute('color', '#2196F3');
  });

  it('should callback after duration end', () => {
    waitFor(() => expect(mockedProps.onLoaded).toHaveBeenCalledTimes(1));
  });
});
