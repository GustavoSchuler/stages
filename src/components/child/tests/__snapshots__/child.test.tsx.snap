// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] Child renders correctly 1`] = `
<div
  class="sc-AxjAm hsszeu"
  data-testid="child__box"
  display="flex"
  height="128"
>
  <div
    class="sc-fzoXWK LcMmt"
    color="#2196F3"
    data-testid="child__loader"
    opacity="0.7"
    size="48"
    speed="2"
  >
    <div />
    <div />
  </div>
</div>
`;
